R/glmClassical.R

```diff
@@ -0,0 +1,1223 @@
+    EMMsummary$addColumnInfo(name = "lowerCI",  title = gettext("Lower"),    type = "number", overtitle = gettextf("%s%% CI", 100 * options[["marginalMeansCiWidth"]]))
+    EMMsummary$addColumnInfo(name = "upperCI",  title = gettext("Upper"),    type = "number", overtitle = gettextf("%s%% CI", 100 * options[["marginalMeansCiWidth"]]))
+  }
+
+  if (options[["marginalMeansComparison"]]) {
+    EMMsummary$addColumnInfo(name = "stat",   title = ifelse(colnames(emmTest)[ncol(emmTest) - 1] == "t.ratio", gettext("t"), gettext("z")), type = "number")
+    EMMsummary$addColumnInfo(name = "pval",   title = gettext("p"),         type = "pvalue")
+    EMMsummary$addFootnote(.EMMmessageTestNull(options[["marginalMeansComparisonWith"]]), colNames = "pval")
+  }
+
+  jaspResults[["emmContainer"]][["EMMsummary"]] <- EMMsummary
+
+  for (i in 1:nrow(emmTable)) {
+    tempRow <- list()
+
+    if (options[["marginalMeansContrast"]])
+      tempRow$Level <- i
+
+
+    for (variable in unlist(options[["marginalMeansVars"]])) {
+
+      if (variable %in% options[["covariates"]])
+        tempRow[variable] <- emmTable[i, variable]
+      else
+        tempRow[variable] <- as.character(emmTable[i, variable])
+
+    }
+
+    # the estimate is before SE (names change for GLM)
+    tempRow$estimate <- emmTable[i, grep("SE", colnames(emmTable)) - 1]
+    tempRow$se       <- emmTable[i, "SE"]
+
+    if (options[["marginalMeansComparison"]]) {
+      tempRow$stat <- emmTest[i, grep("ratio", colnames(emmTest))]
+      tempRow$pval <- emmTest[i, "p.value"]
+    }
+
+    if (options[["marginalMeansCi"]]) {
+      tempRow$lowerCI  <- emmTable[i, ncol(emmTable) - 1]
+      tempRow$upperCI  <- emmTable[i, ncol(emmTable)]
+    }
+
+
+
+    EMMsummary$addRows(tempRow)
+  }
+
+  if (length(emm@misc$avgd.over) != 0)
+    EMMsummary$addFootnote(.EMMmessageAveragedOver(emm@misc$avgd.over))
+
+  # add warning message
+  EMMsummary$addFootnote(ifelse(options[["marginalMeansResponse"]],
+                                gettext("Results are on the response scale."),
+                                gettext("Results are not on the response scale and might be misleading.")))
+
+  object <- list(
+    emm        = emm,
+    emmTable   = emmTable
+  )
+
+  EMMresults$object <- object
+  jaspResults[["emmContainer"]][["EMMresults"]] <- EMMresults
+
+  return()
+}
+
+# Contrast analysis
+.glmContrastsTable <- function(jaspResults, dataset, options, ready, position) {
+
+
+  if (!ready | (length(options[["marginalMeansVars"]]) == 0) | !options[["marginalMeansContrast"]])
+    return()
+
+  if (!is.null(jaspResults[["emmContainer"]][["contrastsTable"]]))
+    return()
+
+  if (is.null(jaspResults[["emmContainer"]][["EMMresults"]]))
+    return()
+
+  emm       <- jaspResults[["emmContainer"]][["EMMresults"]]$object$emm
+  emmTable  <- jaspResults[["emmContainer"]][["EMMresults"]]$object$emmTable
+
+
+  EMMCsummary <- createJaspTable(title = gettext("Contrasts"))
```

Comment:
```suggestion
  contrastsTable <- createJaspTable(title = gettext("Contrasts"))
```
1. follow camelCase, 2. more descriptive name

Reply:
ok.